src/block_iterator.rs

```diff
@@ -50,11 +47,12 @@ pub struct BlockIterator<B: BlockLike> {
     // first key in the block, because slateDB does not support multi version of keys
     // so we use `Bytes` temporarily
     first_key: Bytes,
+    row_attributes: Vec<RowAttribute>,
 }
 
 impl<B: BlockLike> KeyValueIterator for BlockIterator<B> {
     async fn next_entry(&mut self) -> Result<Option<KeyValueDeletable>, SlateDBError> {
-        let Some(key_value) = self.load_at_current_off() else {
+        let Ok(Some(key_value)) = self.load_at_current_off() else {
```

Comment:
Changes in this file are b/c decoding can now throw an error if the encoded row is malformed (i.e. you are reading back data with code that's older than what was encoded with)

Reply:
Should this return (or handle) the error then? Wondering if we should do `self.load_at_current_off()?`.

Reply:
🤦 yes, that was my intention and just forgot to do that